### PR TITLE
fix #12383 typeof returns wrong type for string slice

### LINKED ISSUE #12383: typeof returns wrong type for string slice
According to the tutorial, both ways of specifying the string slice pattern below are the same, but they return a different type when using 'typeof'
(https://nim-lang.github.io/Nim/tut1.html#advanced-types-slices)

### Example
```nim
var b = "some string"
echo $typeof(b)              

echo $typeof(b[0..^1])        
echo $typeof(b[0..b.len-1])   # wrong type
```

### Current Output
```
string
string
char
```

### Expected Output
```
string
string
string
```

A slice pattern that contains "^" will return "string" correctly, but other patterns return "char"

### Additional Information
* Nim Compiler Version 0.20.2

### PR DESCRIPTION
**Pull request by bung87**
fix #12383

### CODE REVIEW

File: tests/misc/t12383.nim [hunk 1]
```diff
@@ -0,0 +1,7 @@
+var b = "some string"
+
+doAssert type(b[0..b.len-1]) is string
+doAssert typeof(b[0..b.len-1]) is string
```
Comment (ringabout) [inline tests/misc/t12383.nim:4]:
```nim
var b = "some string"
doAssert typeof(b[0..b.len-1], typeOfIter) is char
doAssert typeof(b[0..b.len-1], typeOfProc) is string
```
Should be added to test cases too which works before.

Comment (metagn) [reply tests/misc/t12383.nim:4]: Isn't typeOfIter the default? Then the old behavior is correct.

Comment (ringabout) [reply tests/misc/t12383.nim:4]: Yeah, I agree

Comment (bung87) [reply tests/misc/t12383.nim:4]: yeah I misunderstand the `typeof ` design, thought it like other languages do.

**Pull request closed without merging**